superset-frontend/src/components/AnchorLink/index.tsx

```diff
@@ -16,48 +16,32 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { useEffect } from 'react';
 import { t } from '@superset-ui/core';
 
 import URLShortLinkButton from 'src/components/URLShortLinkButton';
 import getDashboardUrl from 'src/dashboard/util/getDashboardUrl';
 import getLocationHash from 'src/dashboard/util/getLocationHash';
 
-const propTypes = {
-  anchorLinkId: PropTypes.string.isRequired,
-  filters: PropTypes.object,
-  showShortLinkButton: PropTypes.bool,
-  inFocus: PropTypes.bool,
-  placement: PropTypes.oneOf(['right', 'left', 'top', 'bottom']),
-};
-
-const defaultProps = {
-  inFocus: false,
-  showShortLinkButton: false,
-  placement: 'right',
-  filters: {},
-};
-
-class AnchorLink extends React.PureComponent {
-  componentDidMount() {
-    const hash = getLocationHash();
-    const { anchorLinkId } = this.props;
+interface AnchorLinkProps {
+  anchorLinkId: string;
+  filters: Record<string, any>;
+  showShortLinkButton?: boolean;
+  inFocus?: boolean;
+  placement: 'right' | 'left' | 'top' | 'bottom';
+}
 
-    if (hash && anchorLinkId === hash) {
-      this.scrollToView();
-    }
-  }
+function AnchorLink({
+  filters = {},
+  showShortLinkButton = false,
+  inFocus = false,
+  placement = "right",
+  ...props
+}: AnchorLinkProps) {
 
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    const { inFocus = false } = nextProps;
-    if (inFocus) {
-      this.scrollToView();
-    }
-  }
```

Comment:
Resources referred to replace the `UNSAFE_componentWillReceiveProps`:

- https://codezup.com/unsafe-componentwillreceiveprops-in-react-hook-functional/
- https://reactjs.org/docs/react-component.html